addon/utils/is-fastboot.js

```diff
@@ -0,0 +1,10 @@
+import { computed } from '@ember/object';
+import { getOwner } from '@ember/application';
+
+export default function isFastBoot() {
+  return computed(function() {
+    const fastboot = getOwner(this).lookup('service:fastboot');
```

Comment:
This CPM must be used on a container-aware object (usually something that the container instantiates, like Service/Route/Controller/Component/Model/Adapter/Serializer, but **NOT** an `Ember.Object.extend({})` that you explicitly create)

We should [assert](https://www.emberjs.com/api/ember/3.1/functions/@ember%2Fdebug/assert) that `getOwner(this)` returns something truthy reasonable
```js
const container = getOwner(this);
assert('You may only use isFastBoot() on a container-aware object',
   container && .lookup === 'function');
```

Reply:
👍